portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -110,6 +111,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       if(numberOfCommentsToDisplay == 0){
         response.getWriter().println(gson.toJson(allBlogMessages)); // set a default amount later.
         return;
+      } else {
+        List<BlogMessage> limitedBlogMessages = new ArrayList<>();
+        for (int i = 0; i < allBlogMessages.size(); i++) {
+          limitedBlogMessages.add(allBlogMessages.get(i));
+        }
+        response.getWriter().println(gson.toJson(limitedBlogMessages));
+        return;
```

Comment:
Both the if and else branches will return, which means the code after line 121 would not be reachable. Is that intended? Normally, unreachable code should be deleted, but I assume you add things here for a quick fix, so I just ask in case.